### ISSUE TITLE
[Package Issue]: Gnu.Emacs

### ISSUE THREAD

**Issue by albertosantini** [microsoft/winget-pkgs#88443]
### Please confirm these before moving forward

- [X] I have searched for my issue and not found a work-in-progress/duplicate/resolved issue.
- [X] I have not been informed if the issue is resolved in a preview version of the winget client.

### Category of the issue

Installation issue.

### Brief description of your issue

After installing Emacs with `winget install GNU.emacs`, the installation was fine and the program works correctly.

Anyway I don't see tha package in the list with `winget list`.
Also I don't see it in `Control Panel / Programs and Features`

### Steps to reproduce

`$ winget install GNU.emacs`

### Actual behavior

`$ winget list`

Tha package is not in the list.

### Expected behavior

The package is in the list.

### Environment

```shell
❯ winget --info
Windows Package Manager v1.3.2691
Copyright (c) Microsoft Corporation. All rights reserved.

Windows: Windows.Desktop v10.0.22621.819
System Architecture: X64
Package: Microsoft.DesktopAppInstaller v1.18.2691.0

Logs: %LOCALAPPDATA%\Packages\Microsoft.DesktopAppInstaller_8wekyb3d8bbwe\LocalState\DiagOutputDir

Links
---------------------------------------------------------------------------
Privacy Statement   https://aka.ms/winget-privacy
License Agreement   https://aka.ms/winget-license
Third Party Notices https://aka.ms/winget-3rdPartyNotice
Homepage            https://aka.ms/winget
Windows Store Terms https://www.microsoft.com/en-us/storedocs/terms-of-sale
```

**Comment by albertosantini** [comment]
Well, installing the package using the installer winget silently executes, the behaviour is the same, intending there is no trace in `Control Panel / Programs and Features`.

Anyway if I install the package from winget, it should show in the list. 
Maybe the missing in the list is a side-effect of the missing in `Programs and Features`

I suppose winget is out of the equation for that.

Sorry for the noise.

**Comment by denelon** [comment]
Yes, that seems to be the case. If the installer doesn't record any entries for Windows Apps & Features, winget has no way to know it's installed.

**Comment by denelon** [comment]
I'll leave this open in the event an issue is created to modify the installer to report its presence on the system.

**Comment by albertosantini** [comment]
Here Emacs bug tracking . I don't see an issue open for this.
https://debbugs.gnu.org/cgi/pkgreport.cgi?package=emacs;include=subject%3Awindows

/cc @Trenly @sootysec 
They are the latest submitters. 
Maybe they are contributing to Emacs community and may be aware about this issue.

**Comment by jypeter** [comment]
I got the same issue with emacs `28.2`, that is:
- emacs does not appear when using the `list` (and `upgrade`) command

I have just discovered today that there was a brand new version of emacs (`29.1`), but of course `upgrade` was still not working
```
PS C:\Users\jypeter> winget show --id GNU.Emacs --versions
Trouvé GNU Emacs [GNU.Emacs]
Version
-------
29.1
28.2
28.1
27.2
27.1

PS C:\Users\jypeter> winget upgrade --id GNU.Emacs
Aucun package installé ne correspond aux critères saisis.
```
I tried _installing_, which worked fine. Trouble is that **I ended up with both  `28.2` and `29.1` installed**, and `winget list` still does not know anything about emacs
```
PS C:\Users\jypeter> winget install --id GNU.Emacs
Trouvé GNU Emacs [GNU.Emacs] Version 29.1
La licence d’utilisation de cette application vous est octroyée par son propriétaire.
Microsoft n’est pas responsable des paquets tiers et n’accorde pas de licences à ceux-ci.
Téléchargement en cours https://ftp.gnu.org/gnu/emacs/windows/emacs-29/emacs-29.1_2-installer.exe
  ██████████████████████████████  70.0 MB / 70.0 MB
Le code de hachage de l’installation a été vérifié avec succès
Démarrage du package d’installation... Merci de patienter.
Installé correctement

PS C:\Users\jypeter> winget list | wsl grep -i emacs

PS C:\Users\jypeter> winget list | wsl grep -i 7zip
7-Zip 23.01 (x64)                        7zip.7zip                                23.01                          winget
```

I ended up uninstalling both versions of emacs, using the `Uninstall` shortcut available from `Start`=>`Emacs-xx.x` and I reinstalled `29.1` using `winget install --id GNU.Emacs` a second time. I now have `29.1` installed, but winget still does not know it's there
```
PS C:\Users\jypeter> winget upgrade --id GNU.Emacs
Aucun package installé ne correspond aux critères saisis.

PS C:\Users\jypeter> winget upgrade --id 7zip.7zip
Mise à niveau disponible introuvable.
Aucune version de package plus récente n’est disponible à partir des sources configurées.
```

I guess **something must indeed be wrong in the emacs installation _configuration_** (or whatever) and still has not been fixed

If anybody knows how to let emacs people know about that...

Still, I think that winget should run some kind of **sanity/consistency check** and report that something must be wrong. In the current case
1. winget was able to find and install something
2. winget was able to detect that the installation was successful because it reports `Installé correctement`
3. **winget should be able to see and report that something is wrong if it does not know anything about something that was just installed correctly**

**Comment by binki** [comment]
@denelon I don’t think this should be added to that milestone because the reported issue isn’t actually fixed.

**Comment by jypeter** [comment]
For the record, I have just googled `"emacs" winget` and the result was not very promising. Most useful hit was a link to this GH issue

It's great if somebody knows a Gnu resource where this issue can be reported, but I think such a reproducible problem is **an even greater opportunity for winget developers** to check what goes wrong and report/fix it! See my _sanity check_ comment above

There is a chance this problem does not occur only when installing emacs, so it's quite worth it investigating what happens on the winget side. Hopefully this is addressed in [1.7 Packages](https://github.com/microsoft/winget-pkgs/milestone/14)

**Comment by jypeter** [comment]
That's great news! Thanks @albertosantini

**Comment by albertosantini** [comment]
@jypeter https://github.com/emacs-mirror/emacs/compare/master...emacs-30
